clients/java/src/main/java/io/camunda/client/impl/worker/JobStreamerImpl.java

```diff
@@ -198,6 +253,19 @@ private void lockedHandleStreamComplete(final Throwable error) {
     }
 
     if (error != null) {
+      if (error instanceof StatusRuntimeException) {
+        final StatusRuntimeException statusError = ((StatusRuntimeException) error);
+        if (statusError.getStatus().getCode() == Status.CANCELLED.getCode()
+            && statusError.getCause() != null
+            && statusError.getCause().getCause() instanceof StreamingTimeoutException) {
```

Comment:
💭 Wait, so it's `getCause().getCause()`? What's the first cause? `CancellationException`?

Reply:
it's `io.grpc.StatusException`, now that you raise it, let's check for that explicitly